testing/tests/DevExpress.ui.widgets.editors/textEditorParts/common.tests.js

```diff
@@ -1384,7 +1384,13 @@ QUnit.module('valueChanged should receive correct event parameter', {
         this.checkEvent = (assert, type, target, key) => {
             const event = this.valueChangedHandler.getCall(0).args[0].event;
             assert.strictEqual(event.type, type, 'event type is correct');
-            assert.strictEqual(event.target, target.get(0), 'event target is correct');
+
+            // NOTE: the cached event.target is missing if the element is in shadow dom
+            // looks like a bug in a browser
```

Comment:
Can we understand is it a browser bug or not?

Reply:
I checked this scenario without devextreme controls and the result was same. We cannot store event data if the element target is in shadow dom. This only affects the test, not production code.